mayastor/src/grpc/mayastor_grpc.rs

```diff
@@ -49,93 +120,282 @@ impl mayastor_server::Mayastor for MayastorSvc {
             return Err(Status::invalid_argument("Missing devices"));
         }
 
-        sync_config(pool_grpc::create(args)).await
+        let rx = rpc_submit::<_, _, LvsError>(async move {
+            let pool = Lvs::create_or_import(args).await?;
+
+            Ok(Pool::from(pool))
+        })?;
+
+        rx.await
```

Comment:
this pattern seems to repeat a lot, could it turned into a fn's or something?

Reply:
Yes, we can but, I want to keep it explicit for now because I want to make it clear that we are going between the two loops.